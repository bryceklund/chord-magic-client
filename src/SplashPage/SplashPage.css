
.intro_header {
    background-color: #27CCBE;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100px;
  }

  .intro_header h1 {
      font-size: 36px;
      animation: animation-hero 1s linear infinite;
      background-size: 800% 800%;
      background-image: repeating-linear-gradient(to left, violet, indigo, blue, green, yellow, orange, red);
      color: transparent;
      -webkit-background-clip: text;
      background-clip: text;
  }

  .intro_header h2 {
      font-size: 12px;
      line-height: 30px;
  }

  .getting_started {
    background-color: #30FFEE;
    height: 150px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  
  .splash_login_links {
    background-color: #3E807A;
    height: 150px;
    text-align: center;
    padding-top: 30px;
    line-height: 40px;
  }

  @media screen and (max-width: 700px) {
    .intro_header h1 {
        font-size: 28px;
    }
    .intro_header h2 {
        font-size: 15px;
    }
    .intro_header {
        height: 150px;
        text-align: center;
    }
  }