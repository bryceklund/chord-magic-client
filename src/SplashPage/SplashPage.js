import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './SplashPage.css';

class SplashPage extends Component {
    render() {
        return (
            <main role='main'>
                <header role='banner' className='intro_header'>
                    <h1>Welcome to Chord Magic!</h1>
                    <h2 className='tagline'>Chord progression building streamlined</h2>
                </header>
                <section className="getting_started">
                    <p className='instructions'>It's great to have you here. If you already know what you're doing, you can&nbsp;<a href='#'>jump to the bottom</a>&nbsp;to proceed.
                    <br /><span className='tagline'> Chord Magic</span> is a tool to help you build and save chord progressions. It's made up of a <span className='bold'>timeline:</span><img className='timeline_photo' src={require('../introduction/timeline.png')} alt='timline' />and a <span className='bold'>library:</span> 
                     You can move through the timline by using the arrows underneath to scub left/right, or jump to the beginning/end. 
                     Select a chord to view its scale, instrument, and octave. Move </p>
                </section>
                <section className="splash_login_links">
                    <Link to='/login' className='splash_login'>Login</Link>&nbsp;&nbsp;
                    <Link to='/register' className="splash_register">Register</Link><br />
                    <Link to='/editor' className="splash_continue">Continue without an account</Link>
                </section>
            </main>
        );
    }
}

export default SplashPage;