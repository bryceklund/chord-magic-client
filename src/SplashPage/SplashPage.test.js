import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import SplashPage from './SplashPage'

describe(`SplashPage`, () => {
    it(`renders without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <BrowserRouter>
                <SplashPage />
            </BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it(`matches its snapshot`, () => {
        const tree = renderer
                            .create(<BrowserRouter>
                                        <SplashPage />
                                    </BrowserRouter>)
                            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})