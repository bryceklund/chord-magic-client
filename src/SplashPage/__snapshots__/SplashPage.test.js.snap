// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashPage matches its snapshot 1`] = `
<main
  role="main"
>
  <header
    className="intro_header"
    role="banner"
  >
    <h1>
      Welcome to Chord Wizard!
    </h1>
    <h2>
      Chord progression building, streamlined.
    </h2>
  </header>
  <section
    className="getting_started"
  >
    Getting started instructions
  </section>
  <section
    className="splash_login_links"
  >
    <a
      className="splash_login"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
      
    <a
      className="splash_register"
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
    <br />
    <a
      className="splash_continue"
      href="/editor"
      onClick={[Function]}
    >
      Continue without an account
    </a>
  </section>
</main>
`;
